frontend/src/pages/LandlordPage.tsx

```diff
@@ -97,56 +56,68 @@ const LandlordPage = (): ReactElement => {
     });
   }, [landlordId, showConfirmation]);
 
+  useEffect(() => {
+    get<Landlord>(`/landlord/${landlordId}`, {
+      callback: setLandlordData,
+    });
+  }, [landlordId]);
+
+  useEffect(() => {
+    get<Apartment[]>(`/buildings/${landlordId}`, {
+      callback: setBuildings,
+    });
+  }, [landlordId]);
+
+  useEffect(() => {
+    if (landlordData && buildings && reviewData) {
+      setLoaded(true);
+    }
+  }, [landlordData, buildings, reviewData]);
+
+  useEffect(() => {
+    return subscribeLikes(setLikedReviews);
+  }, []);
+
   const showConfirmationToast = () => {
     setShowConfirmation(true);
     setTimeout(() => {
       setShowConfirmation(false);
     }, toastTime);
   };
 
-  useEffect(() => {
-    return subscribeLikes(setLikedReviews);
-  }, []);
-
-  const addLike = async (reviewId: string) => {
-    try {
-      const user = await getUser(true);
-      if (!user) {
-        throw new Error('Failed to login');
+  const likeHelper = (dislike = false) => {
+    return async (reviewId: string) => {
+      setLikeStatuses((reviews) => ({ ...reviews, [reviewId]: true }));
+      try {
+        const user = await getUser(true);
+        if (!user) {
+          throw new Error('Failed to login');
+        }
+        const defaultLikes = dislike ? 1 : 0;
+        const offsetLikes = dislike ? -1 : 1;
+        const token = await user.getIdToken(true);
+        const endpoint = dislike ? '/remove-like' : '/add-like';
+        await axios.post(endpoint, { reviewId }, createAuthHeaders(token));
```

Comment:
Didn't catch this in an earlier review, but we should also refactor all instances of `axios.post` to a helper function so that we can easily change the route in the future (this matters if we deploy the server and frontend separately)